### ISSUE TITLE
(aws_cdk): Py module error when using Bazel

### ISSUE THREAD

**Issue by psalvaggio-dl** [aws/jsii#3881]
### Describe the bug

I have been able to localize the issue I am having to version 2.51.0. Prior to this version, all of my pylint tests are passing. However, after this version, I am getting many errors of the following form:
```
E0611: No name 'Duration' in module 'aws_cdk' (no-name-in-module)
```
This seems to apply to everything in the top-level `aws_cdk` package. Subpackage import still seem to be working fine.

### Expected Behavior

Clean pylint on imports like:

```
from aws_cdk import (
    Duration,
)
```

### Current Behavior

```
E0611: No name 'Duration' in module 'aws_cdk' (no-name-in-module)
```

### Reproduction Steps

I'm having difficulty reproducing outside of my monorepo setup, so I'm mainly just looking for any insight on if anything changed and what might have caused this behavior.

### CDK CLI Version

2.53.0

### Framework Version

2.53.0

### Node.js Version

18.6.0

### OS

Ubuntu

### Language

Python

### Language Version

3.9.10

**Comment by psalvaggio-dl** [comment]
This issue also appears to be affecting versions prior to 2.51.0 if alpha modules are used. I can work around these issues by disabling testing on all code that uses alpha modules, but this is not ideal. I am still stuck on 2.50 because with 2.51, the issue propagates up to the main `aws_cdk` module, rendering all CDK code untestable.

**Comment by michaelboyd2** [comment]
It would be really useful if this was fixed - it is essentially preventing Bazel users from using recent aws-cdk-lib versions.

**Comment by randyridgley** [comment]
+1 I have a customer facing this same issue and looking for guidance

**Comment by michaelboyd2** [comment]
> Regarding the changes made by @psalvaggio-dl, it seems like the primary change for getting past the import failures are actually using the `py_test` offered by [aspect_rules_py](https://github.com/aspect-build/rules_py). I believe this is due to how they handle packages in their rules, which mitigates the namespace problem. In our case, it was a matter of adding:
> 
> ```
> http_archive(
>     name = "aspect_rules_py",
>     sha256 = "0cfd8d2492a1a63c3881abd759a54c4bb8a3555cb85971fb8de5cc8315546c42",
>     strip_prefix = "rules_py-846674298d84663dcc109b6b20f5f8fc39aacfd4",
>     url = "https://github.com/aspect-build/rules_py/archive/846674298d84663dcc109b6b20f5f8fc39aacfd4.tar.gz",
> )
> 
> load("@aspect_rules_py//py:repositories.bzl", "rules_py_dependencies")
> ```
> 
> to `WORKSPACE.bazel` and changing the testing imports to:
> 
> ```
> #load("@rules_python//python:defs.bzl", "py_test")
> load("@aspect_rules_py//py:defs.bzl", "py_test")
> ```
> 
> I still think this should be addressed , but I wanted to provide another workaround for anyone else stumbling into this problem.

Thanks, this might be a useful workaround for some people, but it is quite a major change to how Python is being built which many will not want to go with. I only mention this because I do not want an AWS developer who is not that familiar with Bazel to read your post and think "Ah there is an easy workaround, this does not need to be prioritised", because that would not be right. I appreciate this is not what you were trying to say, and it is useful for the option you mention to be made more explicit.

**Comment by hgad** [comment]
Would really appreciate it if someone can point me to a Bazel workspace that has targets for synthesizing and deploying a Python-based CDK. Can't seem to find an example anywhere.

**Comment by scott-osk** [comment]
+1 we are also encountering this issue. @psalvaggio-dl 's fix above seems to have fixed our issue for the meantime.

**Comment by michaelboyd2** [comment]
@psalvaggio-dl did this fix the issue for you? It didn't for me, although I think it might have helped. I was able to get a simple example working with `aws-cdk-lib==2.135.0` with a hack to make sure `import aws_cdk` found the aws_cli_lib rather than `aws_cdk_asset_awscli_v1` (I did this by using `aws_cdk.__file__` to check where it was looking and then altering the `sys.path` in the test).

My understanding is that `aws_cdk_lib`, `aws_cdk_asset_awscli_v1` and a few others are intended to be a single namespace package and Bazel does not understand this. I think in some sense ideally what would happen is the `PYTHONPATH` would point towards `aws_cdk_lib` which would set `__path__` correctly so as to include the other packages?
